### PR TITLE
[2/n][dagster-powerbi] scaffold DagsterPowerBITranslator class

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Builds out a very barebones translator class to build specs for PowerBI assets.

The structure of this translator is slightly different than our existing ones, curious for feedback. Instead of implementing many functions to define each property of an asset, it has a single function that returns an asset spec (which calls out to other functions for each object type).

The user also supplies a translator class rather than a translator instance. The ontology in this model is:
- A translator class defines the logic for how to translate API data into asset specs.
- A user can subclass a translator class to override or customize this logic.
- An instance of a translator class represents a translation "session," provided with any global context information (`PowerBIWorkspaceData`, in this case).
- An invocation of `get_asset_spec` represents translating an individual object, and is provided with local context (the object we're translating, `PowerBIContentData`, in this case).

One hesitation is that `Type[DagsterPowerBITranslator]` is an awkward and potentially confusing type annotation vs passing an instance.

For PowerBI (and I'd imagine many integrations), we need global context to resolve local translation because each local object has relationships with the global context - in this case we need to figure out the asset key of an upstream asset from its internal PowerBI ID.

## Test Plan

n/a, subsequent PRs will test impl - this PR is mostly around settling on the translator API

### CODE REVIEW

**Comment by benpankow** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23245?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#23247** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23247?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23246** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23246?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23245** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23245?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23244** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23244?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-powerbi/dagster_powerbi/translator.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+U = TypeVar("U")
+
+
+# scaffold out what a generic translator class might look like
+class DagsterTranslator(Generic[T, U]):
+    def __init__(self, context: U):
+        self._context = context
+
+    def get_asset_spec(self, data: T) -> AssetSpec:
+        raise NotImplementedError
```
Comment (rexledesma) [inline python_modules/libraries/dagster-powerbi/dagster_powerbi/translator.py:20]: for sake of consistency, im not sure if we should cut over to this pattern right now until other integrations are ready

**Review by rexledesma — CHANGES_REQUESTED** [review]

File: python_modules/libraries/dagster-powerbi/dagster_powerbi/translator.py (hunk 1 above)
Comment (benpankow) [reply python_modules/libraries/dagster-powerbi/dagster_powerbi/translator.py:20]: Yeah, depending on feedback in https://github.com/dagster-io/internal/discussions/10935 may make sense to more closely replicate existing translators.

I do think that an experimental integration with limited users is prob the easiest place to make this initial cutover, that way we can get user feedback before cutting over existing integrations.

**Review by sryza — APPROVED** [review]
We should take out the generic translator piece for now. Otherwise LGTM.

**Pull request merged**